cmd/jaeger/internal/integration/integration.go

```diff
@@ -46,13 +48,23 @@ func (s *StorageIntegration) newDataReceiver(t *testing.T, factories otelcol.Fac
 	cfg, err := configProvider.Get(context.Background(), factories)
 	require.NoError(t, err)
 
+	tel, err := telemetry.New(context.Background(), telemetry.Settings{}, cfg.Service.Telemetry)
+	require.NoError(t, err)
+
 	storageCfg, ok := cfg.Extensions[jaegerstorage.ID].(*jaegerstorage.Config)
 	require.True(t, ok, "no jaeger storage extension found in the config")
 
 	exporterCfg, ok := cfg.Exporters[storageexporter.ID].(*storageexporter.Config)
 	require.True(t, ok, "no jaeger storage exporter found in the config")
 
-	receiver := datareceivers.NewJaegerStorageDataReceiver(exporterCfg.TraceStorage, storageCfg)
+	telemetrySettings := componenttest.NewNopTelemetrySettings()
+	telemetrySettings.Logger = tel.Logger()
+
+	receiver := datareceivers.NewJaegerStorageDataReceiver(
+		telemetrySettings,
```

Comment:
Any reason why can't pass tel directly?

Reply:
Because tel (or `telemetry.Telemetry`) and `component.TelemetrySettings` is two different component.
[telemetry.Telemetry](https://github.com/open-telemetry/opentelemetry-collector/blob/15201f1e5967fd1bc1e51c118178ab109170fe07/service/telemetry/telemetry.go#L61) initialize the `zap.Logger` from the config.
[component.TelemetrySettings](https://github.com/open-telemetry/opentelemetry-collector/blob/15201f1e5967fd1bc1e51c118178ab109170fe07/component/telemetry.go#L20) is the struct that contains `zap.Logger` that is used by components in OTEL to report telemetry matters.

This means that the extension and receiver in the data receiver are using `component.TelemetrySettings` to write logs. Based on your suggestion, I thought it is better if we pass the initialized struct and use it directly in the data receiver.

Reply:
Got it. Looking through OTEL code the situation seems to be quite messy, I don't know if they have a ticket to clean it up. E.g.
  * why are `Telemetry` and `TelemetrySettings` two different structs? They seem to server the same purpose (the latter being more comprehensive)
  * `TelemetrySettings` comment also mentions yet another similar struct `servicetelemetry.TelemetrySettings`